### ISSUE TITLE
N805 First argument of a method should be named self -- false positive for __new__

### ISSUE THREAD

**Issue by LefterisJP** [charliermarsh/ruff#620]
In python `__new__` is a special method and its signature does not need to start with an argument named `self`. It most commonly looks like this:

```python
def __new__(cls, *args, **kwargs)
    ....
```

Yet ruff will complain with `N805 First argument of a method should be named self`

**Comment by charliermarsh** [comment]
(Agreed!)

**Comment by andersk** [comment]
Other exemptions present in the pep8-naming [source](https://github.com/PyCQA/pep8-naming/blob/main/src/pep8ext_naming.py): `__new__`, `__init_subclass__`, `__class_getitem__`, and any method in a class with `type` or `ABCMeta` in its base classes, are automatically class methods.

**Comment by charliermarsh** [comment]
Will fix all of those today.

**Comment by LefterisJP** [comment]
Thank you Charlie. Realy.
